cmd/otelcol/main.go

```diff
@@ -169,66 +161,99 @@ func checkRuntimeParams() {
-func setConfig() {
-	// Check if the config is specified via the env var.
-	config := os.Getenv(configEnvVarName)
-	// If not attempt to use a default config; supports Docker and local
-	if config == "" {
-		_, err := os.Stat(defaultDockerSAPMConfig)
-		if err == nil {
-			config = defaultDockerSAPMConfig
+// Sets flag '--config' to specified env var SPLUNK_CONFIG, if the flag not specified.
+// Logs a message and returns if env var SPLUNK_CONFIG_YAML specified, and '--config' and SPLUNK_CONFIG no specified.
+// Sets '--config' to default config file path if '--config', SPLUNK_CONFIG and SPLUNK_CONFIG_YAML not specified.
+func checkConfig() {
+	configPathFlagExists, configPathFlag := getKeyValue(os.Args[1:], "--config")
+	configPathVar := os.Getenv(configEnvVarName)
+	configYaml := os.Getenv(configYamlEnvVarName)
+
+	if configPathFlagExists && configPathFlag == "" {
+		log.Fatal("Command line flag --config specified but empty")
+	}
+
+	if configPathFlag != "" {
+		if _, err := os.Stat(configPathFlag); err != nil {
+			log.Fatalf("Unable to find the configuration file (%s) ensure flag '--config' is set properly", configPathFlag)
 		}
-		_, err = os.Stat(defaultLocalSAPMConfig)
-		if err == nil {
-			config = defaultLocalSAPMConfig
+
+		if configPathVar != "" && configPathVar != configPathFlag {
+			log.Printf("Both environment variable %v and flag '--config' were specified. Using the flag value %s and ignoring the environment variable value %s in this session", configEnvVarName, configPathFlag, configPathVar)
 		}
-		if config == "" {
-			log.Fatalf("Unable to find the default configuration file, ensure %s environment variable is set properly", configEnvVarName)
+
+		if configYaml != "" {
+			log.Printf("Both environment variable %s and flag '--config' were specified. Using the flag value %s and ignoring the environment variable in this session", configYamlEnvVarName, configPathFlag)
 		}
-	} else {
-		// Check if file exists.
-		_, err := os.Stat(config)
-		if err != nil {
-			log.Fatalf("Unable to find the configuration file (%s) ensure %s environment variable is set properly", config, configEnvVarName)
+
+		log.Printf("Set config to %v", configPathFlag)
+		return
+	}
+
+	if configPathVar != "" {
+		if _, err := os.Stat(configPathVar); err != nil {
+			log.Fatalf("Unable to find the configuration file (%s) ensure %s environment variable is set properly", configPathVar, configEnvVarName)
+		}
+
+		os.Args = append(os.Args, "--config="+configPathVar)
+
+		if configYaml != "" {
+			log.Printf("Both %s and %s were specified. Using %s environment variable value %s for this session", configEnvVarName, configYamlEnvVarName, configEnvVarName, configPathVar)
 		}
+
+		log.Printf("Set config to %v", configPathVar)
+		return
+	}
+
+	if configYaml != "" {
+		log.Printf("Using environment variable %s for configuration", configYamlEnvVarName)
+		return
+	}
+
+	defaultConfigPath := getExistingDefaultConfigPath()
+	validateDefaultConfig(defaultConfigPath)
```

Comment:
https://github.com/signalfx/splunk-otel-collector/pull/462#discussion_r661566937

required env var validation now only occurs if no config arg or env var are used, which is a functional change.  The collector service sets the env var w/ the default value so `validateDefaultConfig(configPathVar)` should be used above.  I think `validateDefaultConfig(configPathFlag)` should be checked as well to preserve existing functionality.

Reply:
Yes, I changed the name of the function to checkRequiredEnvVars() and checked both configPathVar and configPathFlag